### PR TITLE
[RW-5609][risk=medium] Upgrade to java 11

### PR DESCRIPTION
**Pull request by yonghaoy**
Description:
https://stackoverflow.com/questions/44832036/google-app-engine-error-fetch-in-a-thread-that-is-neither-the-original-request

https://stackoverflow.com/questions/53972024/java-lang-noclassdeffounderror-com-google-appengine-api-urlfetch-httpmethod

---
**PR checklist**

- [ ] This PR meets the Acceptance Criteria in the JIRA story
- [ ] The JIRA story has been moved to Dev Review
- [ ] This PR includes appropriate unit tests
- [ ] I have run and tested this change locally
- [ ] I have run the E2E tests on ths change against my local UI and/or API server with `yarn test-local` or [yarn test-local-devup](https://github.com/all-of-us/workbench/blob/main/e2e/README.md#examples)
- [ ] If this includes a UI change, I have taken screen recordings or screenshots of the new behavior and notified the PO and UX designer
- [ ] If this includes an API change, I have updated the appropriate Swagger definitions and notified API consumers
- [ ] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later

### CODE REVIEW

File: api/build.gradle [hunk 1]
```diff
@@ -565,24 +559,26 @@ appengine {  // App Engine tasks configuration
...
+    stage {
+        appEngineDirectory = "src/main/webapp/WEB-INF"
+        extraFilesDirectories = "src/main/webapp"
+    }
+}
+
+bootRun {
+    environment = [
+        'GOOGLE_APPLICATION_CREDENTIALS': file("$rootDir/sa-key.json").getAbsolutePath(),
+        'WORKBENCH_PROPERTY_IS_LOCAL': 'true'
```
Comment (calbach) [inline api/build.gradle:580]: Still needed?

File: api/src/main/java/org/pmiops/workbench/utils/mappers/WorkspaceMapper.java [hunk 2]
```diff
@@ -65,7 +65,8 @@ default WorkspaceResponse toApiWorkspaceResponse(
     return toApiWorkspaceResponse(
         toApiWorkspace(dbWorkspace, firecloudWorkspaceResponse.getWorkspace()),
         firecloudWorkspaceResponse.getAccessLevel());
-  };
+  }
+  ;
```
Comment (calbach) [inline api/src/main/java/org/pmiops/workbench/utils/mappers/WorkspaceMapper.java:69]: nit: rm newline

File: api/src/main/webapp/WEB-INF/app.yaml.template [hunk 3]
```diff
@@ -0,0 +1,29 @@
+service: api
+runtime: java11
+env_variables:
+  GAE_PROFILER_MODE: "cpu,heap"
+  JAVA_USER_OPTS: "-Djava.util.logging.config.file=webapps/root/WEB-INF/logging.properties"
+  SPRING_DATASOURCE_DRIVER_CLASS_NAME: "${DB_DRIVER}"
```
Comment (calbach) [inline api/src/main/webapp/WEB-INF/app.yaml.template:6]: these are now being specified in both `application.properties` and here inthe env vars. Do we need both?

**Review by calbach — COMMENTED** [review]
Overall, changes look reasonable

File: api/src/main/webapp/WEB-INF/app.yaml.template (hunk 3 above)
Comment (yonghaoy) [reply api/src/main/webapp/WEB-INF/app.yaml.template:6]: Are all env vars can be passed into our app's docker? If yes, then we don't need them. 
Let me quickly try without those values

File: api/src/main/java/org/pmiops/workbench/monitoring/StackdriverStatsExporterService.java [hunk 4]
```diff
@@ -121,28 +117,28 @@ private String getLocation() {
   }
 
   private String getNodeId() {
-    try {
-      return modulesService.getCurrentInstanceId();
-    } catch (ModulesException e) {
+    // See https://cloud.google.com/appengine/docs/standard/java11/runtime#environment_variables
+    Optional<String> nodeId = AppEngineConfig.getGaeNodeId();
+    logger.log(Level.INFO, String.format("GAE_INSTANCE is %s.", nodeId));
```
[the remark this answers is not in the document]
Comment (yonghaoy) [reply api/src/main/java/org/pmiops/workbench/monitoring/StackdriverStatsExporterService.java:122]: Seems this methoed is only invoked from createAndRegister
And that one is only called during initialization https://source.corp.google.com/search?q=file:%5Eall-of-us%2F%20package:%5Egithub$%20%20createAndRegister&sq=

**Pull request closed without merging**